.github/workflows/release.yml

```diff
@@ -0,0 +1,326 @@
+        id: aptInstall
+        run: sudo apt-get -y install build-essential libpam-dev debhelper devscripts fakeroot
+      - name: Build
+        id: build
+        run: |
+          PATH=$PATH:/usr/local/go/bin:/home/admin/go/bin
+          make linux-artifacts
+      - name: Upload Release Assets
+        id: upload-release-assets
+        run: |
+          tag_name="${GITHUB_REF##*/}"
+          hub release edit $(find ./.releases -type f -printf "-a %p ") -m "" "$tag_name"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  build_upload_windows_assets:
+    name: Build & Upload Windows Assets
+    runs-on: ubuntu-latest
+    needs: create_release
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '^1.15.6'
+      - name: Build
+        id: build
+        run: |
+          PATH=$PATH:/usr/local/go/bin:/home/admin/go/bin
+          make windows-artifacts
+      - name: Upload Release Assets
+        id: upload-release-assets
+        run: |
+          tag_name="${GITHUB_REF##*/}"
+          hub release edit $(find ./.releases -type f -printf "-a %p ") -m "" "$tag_name"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  build_upload_docker:
+    name: Build & Upload Docker Images
+    runs-on: ubuntu-latest
+    needs: create_release
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '^1.15.6'
+      - name: Build
+        id: build
+        run: |
+          PATH=$PATH:/usr/local/go/bin:/home/admin/go/bin
+          make docker-artifacts
+        env:
+          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+          DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+
+  build_upload_aws_s3_binaries:
+    name: Build & Upload AWS S3 Binaries
+    runs-on: ubuntu-latest
+    needs: create_release
+    if: ${{ needs.create_release.outputs.is_prerelease }} == 'false'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '^1.15.6'
+      - name: Build
+        id: build
+        run: |
+          PATH=$PATH:/usr/local/go/bin:/home/admin/go/bin
+          make -j1 binary-linux binary-darwin binary-windows
+          mkdir -p ./.releases
+          cp ./output/binary/linux/bin/step ./.releases/step-linux-${{ needs.create_release.outputs.version }}
+          cp ./output/binary/linux/bin/step ./.releases/step-linux-latest-integration
+          cp ./output/binary/darwin/bin/step ./.releases/step-darwin-${{ needs.create_release.outputs.version }}
+          cp ./output/binary/windows/bin/step ./.releases/step-windows-${{ needs.create_release.outputs.version }}.exe
+      - name: Upload s3
+        id: upload-s3
+        uses: jakejarvis/s3-sync-action@master
```

Comment:
Should we clone non-github actions?

Reply:
Full list: 

jakejarvis/s3-sync-action@master -- https://github.com/jakejarvis/s3-sync-action (500 stars)
prewk/s3-cp-action@master -- https://github.com/prewk/s3-cp-action (fork of s3-sync-action)
smallstep/pkgbuild-action@v1.0.2 -- https://github.com/smallstep/pkgbuild-action
ad-m/github-push-action@master -- https://github.com/ad-m/github-push-action (500 stars)
codecov/codecov-action@v1 -- https://github.com/codecov/codecov-action (550 stars)

For the ones with less stars I have done a code review.

We _could_ fork them all (like I did with the pkgbuild-action one). Depends on what we feel is safe. 
If these were already supported then I would have used the ones from github, but they're not in their lib.

Reply:
We can also pin to a specific SHA

Reply:
🤷‍♂️ pinning to a specific SHA would be safer, popular repos can become an attack vector. I don't think GitHub can do anything about that.